circuit/program/src/data/plaintext/equal.rs

```diff
@@ -32,7 +32,18 @@ impl<A: Aleo> Equal<Self> for Plaintext<A> {
                 }
                 false => Boolean::constant(false),
             },
-            (Self::Literal(..), _) | (Self::Struct(..), _) => Boolean::constant(false),
+            (Self::Array(a, _), Self::Array(b, _)) => match a.len() == b.len() {
+                true => {
+                    // Recursively check each element for equality.
+                    let mut equal = Boolean::constant(true);
+                    for (plaintext_a, plaintext_b) in a.iter().zip_eq(b.iter()) {
+                        equal &= plaintext_a.is_equal(plaintext_b);
+                    }
+                    equal
```

Comment:
```suggestion
                    if a.iter().zip_eq(b.iter()).all(|(plaintext_a, plaintext_b)| plaintext_a.is_equal(plaintext_b))
                        Boolean::constant(true);
                    } else {
                        Boolean::constant(false);
                    }
```
for short-circuiting